core/domain/draft_upgrade_services.py

```diff
@@ -134,89 +134,107 @@ def _convert_html_in_draft_change_list(
+                state_content = cast(state_domain.SubtitledHtmlDict, new_value)
+                state_content['html'] = conversion_fn(state_content['html'])
+            elif (edit_state_property_cmd.property_name ==
                   exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
-                # Only customization args with the key 'choices' have HTML
-                # content in them.
-                # Ruling out the possibility of any other type for MyPy type
-                # checking because here 'new_value' is a value of attribute
-                # that was dynamically defined on change_domain and that
-                # attribute has str type. Thus to convert the type, we used
-                # assert here.
-                assert isinstance(new_value, dict)
-                if 'choices' in new_value.keys():
+                # Here we use cast because in this 'elif clause' we are
+                # updating the interaction's customization args of a State
+                # which can only be of type Dict[str, Dict[str, Any]]. So,
+                # to rule out all other property types for MyPy type checking,
+                # we used cast here.
+                interaction_cust_arg = cast(
+                    Dict[str, Dict[str, Any]], new_value
+                )
+                if 'choices' in interaction_cust_arg.keys():
                     for value_index, value in enumerate(
-                            new_value['choices']['value']):
+                            interaction_cust_arg['choices']['value']):
                         if isinstance(value, dict) and 'html' in value:
-                            new_value['choices']['value'][value_index][
-                                'html'
-                            ] = conversion_fn(value['html'])
+                            interaction_cust_arg['choices']['value'][
+                                value_index
+                            ]['html'] = conversion_fn(value['html'])
                         elif isinstance(value, str):
-                            new_value['choices']['value'][value_index] = (
-                                conversion_fn(value))
-            elif (change.property_name ==
+                            interaction_cust_arg['choices']['value'][
+                                value_index
+                            ] = conversion_fn(value)
+            elif (edit_state_property_cmd.property_name ==
                   exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
-                # Ruling out the possibility of any other type for MyPy
-                # type checking, because here 'new_value' is a value of
-                # attribute that was dynamically defined on change_domain
-                # and that attribute has str type. Thus to convert the
-                # type, we used assert here.
-                assert isinstance(new_value, dict)
+                # Here we use cast because in this 'elif clause' we are
```

Comment:
Same notes for the others.

Reply:
New subtypes are defined! now instead of casting to a particular type, we are casting to cmd type.